R/sql-context.R

```diff
@@ -0,0 +1,190 @@
+# The env for storing rolling_group_by methods
+polars_sql_context__methods <- new.env(parent = emptyenv())
+
+#' @export
+wrap.PlRSQLContext <- function(x) {
+  self <- new.env(parent = emptyenv())
+  self$`_ctxt` <- x
+
+  class(self) <- c("polars_sql_context", "polars_object")
+  self
+}
+
+get_frame_locals <- function(all_compatible) {
```

Comment:
Currently this only looks in the global env and not in all parent frames as in py-polars: https://github.com/pola-rs/polars/blob/main/py-polars/polars/_utils/various.py#L499

(`_get_stack_locals()` is called from `_get_frame_locals()`)

I found it surprisingly difficult to collect all objects in all parent frames without running into recursion issues.